thanosql/_base_client.py

```diff
@@ -125,14 +125,30 @@ def _request(
                     message = response_json["error"].get("message", message)
 
                 if code in {400, 405, 422}:
-                    raise thanosql_error.ThanoSQLValueError(message=message)
+                    raise thanosql_error.ThanoSQLValueError(
+                        message=f"Invalid input: {message}"
+                    )
                 elif code in {401, 403}:
-                    raise thanosql_error.ThanoSQLPermissionError(message=message)
+                    raise thanosql_error.ThanoSQLPermissionError(
+                        message=f"Operation not permitted: {message}"
+                    )
                 elif code == 404:
-                    raise thanosql_error.ThanoSQLNotFoundError(message=message)
+                    raise thanosql_error.ThanoSQLNotFoundError(
+                        message=f"Not found: {message}"
+                    )
                 elif code == 409:
-                    raise thanosql_error.ThanoSQLAlreadyExistsError(message=message)
+                    raise thanosql_error.ThanoSQLAlreadyExistsError(
+                        message=f"Object already exists: {message}"
+                    )
+                elif code == 413:
+                    raise thanosql_error.ThanoSQLConnectionError(
+                        message=f"Entity size too large: {message}"
+                    )
                 elif code == 500:
+                    raise thanosql_error.ThanoSQLInternalError(
+                        message=f"Internal server error: {message}"
+                    )
+                else:
```

Comment:
Added an else case, just in case another error which code is not caught here happens.

Question: I'm not sure whether we need to include the HTTP error code in the error message. It doesn't look important in most cases except when an 'unknown' error happens -> like the 413 error, this 'else' error will be very difficult to debug as we don't know what the HTTP error code is.
Is it better to create a short intro for this case so users are not too clueless, like the others? But I don't know what it should be...